OpenRA.Mods.Common/Traits/DamagedByTerrain.cs

```diff
@@ -70,7 +74,7 @@ public void AddedToWorld(Actor self)
 			// Actors start with maximum damage applied
 			var delta = health.HP - damageThreshold;
 			if (delta > 0)
-				health.InflictDamage(self, self.World.WorldActor, new Damage(delta), false);
+				self.InflictDamage(self, new Damage(delta, Info.DamageTypes));
```

Comment:
This isn't a suicide so I think you should revert this back to:

```csharp
self.InflictDamage(self.World.WorldActor, new Damage(Info.Damage, Info.DamageTypes));
```

I checked. It doesn't have a negative impact. Still fixes https://github.com/OpenRA/OpenRA/issues/11383.